easybuild/framework/easyblock.py

```diff
@@ -663,6 +667,52 @@ def obtain_file(self, filename, extension=False, urls=None, download_filename=No
                 if self.dry_run:
                     self.dry_run_msg("  * %s found at %s", filename, foundfile)
                 return foundfile
+            elif git_config:
+                # if a non-empty dictionary was provided, download from git and archive
+                if not isinstance(git_config, dict):
+                    raise EasyBuildError("Found a non null git_config in 'sources', but value is not a dictionary")
+                else:
+                    git_config = git_config.copy()
```

Comment:
Why the `copy()`? I don't see the point of that...

Reply:
I took this strategy from similar code above (line 347).

Reply:
Ah, now it clicks... It's done to avoid modifying the value that is passed into `obtain_file` with the `.pop`s...

Better add a comment to explain that (on line 347 too, while you're at it). ;)

Reply:
done